### ISSUE TITLE
Unique constraint error when activating repo which was previously deleted

### ISSUE THREAD

**Issue by lonix1** [woodpecker-ci/woodpecker#2315]
### Component

server

### Describe the bug

Repro:
- I had a repo in gitea, and activated it in woodpecker
- I deleted the repo in gitea and deleted it in woodpecker
- I added a repo in gitea, with the same org and name
- I tried to activate it in woodpecker
- Result was a small red popup box on the bottom right:
  > UNIQUE constraint failed: repos.repo_owner, repos.repo_name

I changed logging from info to trace, but there is no error in the logs. I tried restarting the woodpecker server but that didn't help.

This scenario happens with automated provisioning and testing of our servers and services.

Is there any extra info I can supply? Or something I can test?

### System Info

```shell
1.0.2
docker
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by 6543** [comment]
my guess: on repo deletion we don't get rid of all related things :/

**Comment by mzampetakis** [comment]
Tried to repro this using the v1.0.2 and the [[latest main](https://github.com/woodpecker-ci/woodpecker/commit/503252df38154a1ec78dd1371f415018fbec6116)](https://github.com/woodpecker-ci/woodpecker/commit/503252df38154a1ec78dd1371f415018fbec6116) and bitbucket & github (was easier for me as I had already this set up) but with no luck.

I followed these steps:

1. I had a repo in bitbucket, and activated it in woodpecker ✅
2. I deleted the repo in bitbucket and deleted it (manually) in woodpecker ✅ (at this point there was no entry at the `repos` DB table)
3. I added a repo in bitbucket, with the same org and name ✅
4. I tried to activate it in woodpecker ✅

Could this be related somehow only to gitea? I will try to repro using gitea as well.

Repos table has as UNIQUE key the `name` column. This is valid as woodpecker-ci currently supports single repo but should not cause any error at this case where the repo is completely removed from the DB upon deletion.
